src/storage/heap_file.c

```diff
@@ -20347,23 +20140,23 @@ heap_mvcc_check_and_lock_for_delete (THREAD_ENTRY * thread_p, const OID * oid, O
 
       /* Prepare for getting record again. Since pages have been unlatched, others may have changed them (even our
        * record). */
-      if (heap_prepare_get_record (thread_p, oid, class_oid, forward_oid, home_page_watcher, fwd_page_watcher,
-				   record_type, PGBUF_LATCH_READ, false, LOG_WARNING_IF_DELETED) != S_SUCCESS)
+      heap_init_get_context (&context, oid, class_oid, NULL);
+      if (heap_get_prepare_context (thread_p, &context, PGBUF_LATCH_READ, false, LOG_WARNING_IF_DELETED) != S_SUCCESS)
 	{
 	  goto error;
 	}
       /* Make sure everything is as expected. */
-      assert (*record_type == REC_HOME || *record_type == REC_RELOCATION || *record_type == REC_BIGONE);
+      //assert (*record_type == REC_HOME || *record_type == REC_RELOCATION || *record_type == REC_BIGONE);
 
       /* Check page pointer and oid match. */
-      assert (HEAP_IS_PAGE_OF_OID (home_page_watcher->pgptr, oid));
+      //assert (HEAP_IS_PAGE_OF_OID (home_page_watcher->pgptr, oid));
       /* Make sure that forward_oid and forward page are obtained if rec_type is REC_BIGONE or REC_RELOCATION. */
-      assert (*record_type == REC_HOME
-	      || (!OID_ISNULL (forward_oid) && HEAP_IS_PAGE_OF_OID (fwd_page_watcher->pgptr, forward_oid)));
+      //assert (*record_type == REC_HOME || (!OID_ISNULL (forward_oid) && HEAP_IS_PAGE_OF_OID (fwd_page_watcher->pgptr, forward_oid)));
 
       /* Get MVCC header */
-      if (heap_get_mvcc_header (thread_p, oid, forward_oid, home_page_watcher->pgptr, fwd_page_watcher->pgptr,
-				*record_type, recdes_header) != S_SUCCESS)
+      if (heap_get_mvcc_header
+	  (thread_p, oid, &context.forward_oid, context.home_page_watcher.pgptr, context.fwd_page_watcher.pgptr,
+	   context.record_type, recdes_header) != S_SUCCESS)
```

Comment:
@andrei14vl 

It would be nice to change the code layout. 

```C
if (heap_get_mvcc_header (thread_p, ...
```
It helps cscope to correctly understand it is a function call.